beamer/tests/contracts/test_request_manager.py

```diff
@@ -675,6 +751,64 @@ def test_withdraw_without_challenge_with_resolution(
         request_manager.withdraw(claim_id, {"from": claimer})
 
 
+def test_withdraw_two_challengers(
+    request_manager, token, claim_stake, finalization_time, challenge_period_extension
+):
+    claimer, first_challenger, second_challenger, requester = alloc_accounts(4)
+    request_id = make_request(request_manager, token, requester, requester, 1)
+    claim = request_manager.claimRequest(request_id, 0, {"from": claimer, "value": claim_stake})
+    claim_id = claim.return_value
+    first_challenger_eth_balance = web3.eth.get_balance(first_challenger.address)
+    second_challenger_eth_balance = web3.eth.get_balance(second_challenger.address)
+
+    # First challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": first_challenger, "value": claim_stake + 1})
+    # Claimer outbids again
+    request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 10})
+    # Second challenger challenges
+    request_manager.challengeClaim(
+        claim_id, {"from": second_challenger, "value": claim_stake + 11}
+    )
+
+    # Withdraw must fail when claim period is not over
+    with brownie.reverts("Claim period not finished"):
+        request_manager.withdraw(claim_id, {"from": first_challenger})
+    # Withdraw must fail when claim period is not over
+    with brownie.reverts("Claim period not finished"):
+        request_manager.withdraw(claim_id, {"from": second_challenger})
+    # Withdraw must fail when claim period is not over
+    with brownie.reverts("Claim period not finished"):
+        request_manager.withdraw(claim_id, {"from": claimer})
+
+    # Timetravel after claim period
+    chain.mine(timedelta=finalization_time + challenge_period_extension)
+
+    # Show that last challenger can withdraw first
+    request_manager.withdraw(claim_id, {"from": second_challenger})
+
+    # Challenger cannot withdraw twice
+    with brownie.reverts("Challenger has nothing to withdraw"):
+        request_manager.withdraw(claim_id, {"from": second_challenger})
+    with brownie.reverts("Challenger has nothing to withdraw"):
+        request_manager.withdraw(claim_id, {"from": claimer})
+
+    # First challenger withdraws his partial stake
+    request_manager.withdraw(claim_id, {"from": first_challenger})
+
+    assert (
+        web3.eth.get_balance(first_challenger.address)
+        == first_challenger_eth_balance + claim_stake + 1
+    )
+    assert (
+        web3.eth.get_balance(second_challenger.address)
+        == second_challenger_eth_balance + claim_stake + 9
+    )
```

Comment:
I am not sure whether this is already covered, but it would be nice to 
- test that the withdraw order does not matter, i.e. if the fist challenger withdraws first, the amounts withdrawn are still the same
- test that if the claimer is last and wins, all stakes go to the claimer (we have a single-challenger test for this, but I'm not sure for multi-party bidding)

Reply:
the second bullet point is covered

Reply:
can you elaborate what exactly you want to test ? What do you mean by the amounts withdrawn are the same?

Reply:
> can you elaborate what exactly you want to test ? What do you mean by the amounts withdrawn are the same?

I would like to test that for
```
challenger1 withdraws first, gets A ETH
challenger2 withdraws second, gets B ETH
```
and
```
challenger2 withdraws first, gets C ETH
challenger1 withdraws second, gets D ETH
```
the following holds:
```
A == D
B == C
```

Reply:
added test for this